### PR TITLE
WIP: docs: Document build process

### PR DESCRIPTION
**Pull request by AntonioND**
This is an initial draft of the documentation of the build process. Let me know if I'm forgetting something. I still have to review it and improve it, it's not a final version.

I've also added a FAQ.

Thanks to @asiekierka for collecting the list of questions that I used as starting point for this.

### CODE REVIEW

File: docs/content/technical/build_process.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+---
+title: 'Build process information'
+weight: 35
+---
+
+## 1. Introduction
+
+NDS ROMs are complicated. They have an ARM9 binary, an ARM7 binary, an optional
```
Comment (asiekierka) [inline docs/content/technical/build_process.md:8]: Writing "NDS ROMs are complicated." as the first sentence is going to be discouraging to people :-) Don't make them feel condescended to.

File: docs/content/technical/build_process.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+This part is pretty standard. The developer provides source code files that
+are compiled and linked into an `.elf` file (one per CPU). The source code files
+may contain code or data as requested by the developer. This code and data is
+placed into different sections as with any other `.elf` file:
+
+- `text`: Code.
+- `rodata`: Constant data.
+- `data`: Initialized variables.
+- `bss`: Uninitialized variables (they start as 0).
```
Comment (asiekierka) [inline docs/content/technical/build_process.md:27]: There are a few other notable sections provided by our link scripts:

- ITCM (`.vectors` `.itcm`)/DTCM (`.dtcm` `.sbss`) sections
- Secure area (`.secure`) - space reserved for the secure area, which DS/DSi homebrew does not encrypt or use

File: docs/content/technical/build_process.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+MPU (Memory Protection Unit) of the ARM9 needs to be setup correctly, and it is
+needed to ensure that things like timers, DMA, etc, are set to an idle state.
+Unfortunately, this also means that the `crt0` depends on `libnds`, which is the
+one that contains the hardware initialization code.
+
+All of this means that invoking `gcc` isn't as simple as running `gcc file.c`.
+It also means that `gcc` updates may require small changes to the command line
+options used. Some of this is hidden from the user in `specs` files located
+[here](https://github.com/blocksds/sdk/tree/master/sys/crts). This is done
+purely for convenience. You can skip the `specs` files and copy the build
```
Comment (asiekierka) [inline docs/content/technical/build_process.md:59]: for end user convenience and to reduce the amount of breaking changes to user project Makefiles

File: docs/content/technical/build_process.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+- One ROM icon (which may be animated for DSi-compatible games). This icon is
+  displayed in the NDS firmware menu. It's optional.
+
+`ndstool` reads the `elf` file of each CPU, takes every section in the `elf`
+file, and saves it to the NDS ROM as binary data. This includes `twl` sections
+as well as regular sections.
+
+It is possible to generate a NDS ROM without any DSi code or data by running
+`ndstool` with `-h 0x200`, which will force the NDS ROM header to be DS-only. In
+this case, none of the the `twl` sections will be added to the final NDS ROM.
```
Comment (asiekierka) [inline docs/content/technical/build_process.md:115]: This option also generates a PassMe header; while initially intended for loading DS homebrew from Slot-2 cartridges, it is also used by some outdated homebrew application loaders.

File: docs/content/technical/build_process.md (hunk 1 above)
Comment (AntonioND) [reply docs/content/technical/build_process.md]: FAir

**Comment by AntonioND** [comment]
I've merged this, but it's still WIP.

**Pull request merged**